lib/fizzy/execute.cpp

```diff
@@ -744,24 +752,24 @@ execution_result execute(
         }
         case Instr::end:
         {
+            // When we reach end instruction normally, labels have one implicit function block.
```

Comment:
Still a bit of a special case remains here - in one case we pop implicit block and go to end, in another just go to end.